src/transport/SessionManager.cpp

```diff
@@ -573,20 +579,162 @@ void SessionManager::OnMessageReceived(const PeerAddress & peerAddress, System::
         }
         else
         {
-            SecureUnicastMessageDispatch(partialPacketHeader, peerAddress, std::move(msg));
+            SecureUnicastMessageDispatch(partialPacketHeader, peerAddress, std::move(msg), ctxt);
         }
     }
     else
     {
-        UnauthenticatedMessageDispatch(partialPacketHeader, peerAddress, std::move(msg));
+        UnauthenticatedMessageDispatch(partialPacketHeader, peerAddress, std::move(msg), ctxt);
+    }
+}
+
+#if INET_CONFIG_ENABLE_TCP_ENDPOINT
+void SessionManager::HandleConnectionReceived(Transport::ActiveConnectionState * conn)
+{
+    Transport::AppConnCallbackCtxt * appConnCbCtxt = nullptr;
+    char peerAddrBuf[chip::Transport::PeerAddress::kMaxToStringSize];
+
+    VerifyOrReturn(conn != nullptr);
+
+    appConnCbCtxt = conn->mAppState;
```

Comment:
Why is this not just declared here.  Please declare things where used, not at the tops of methods.

Reply:
Done.